src/main/java/org/opensearch/knn/index/util/Lucene.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.util;
+
+import com.google.common.collect.ImmutableMap;
+import org.apache.lucene.util.Version;
+import org.opensearch.knn.index.KNNMethod;
+import org.opensearch.knn.index.KNNSettings;
+import org.opensearch.knn.index.MethodComponent;
+import org.opensearch.knn.index.Parameter;
+import org.opensearch.knn.index.SpaceType;
+
+import java.util.Map;
+
+import static org.opensearch.knn.common.KNNConstants.METHOD_HNSW;
+import static org.opensearch.knn.common.KNNConstants.METHOD_PARAMETER_EF_CONSTRUCTION;
+import static org.opensearch.knn.common.KNNConstants.METHOD_PARAMETER_M;
+
+/**
+ * KNN Library for Lucene
+ */
+public class Lucene extends JVMLibrary {
+
+    final static Map<String, KNNMethod> METHODS = ImmutableMap.of(
+        METHOD_HNSW,
+        KNNMethod.Builder.builder(
+            MethodComponent.Builder.builder(METHOD_HNSW)
+                .addParameter(
+                    METHOD_PARAMETER_M,
+                    new Parameter.IntegerParameter(METHOD_PARAMETER_M, KNNSettings.INDEX_KNN_DEFAULT_ALGO_PARAM_M, v -> v > 0)
```

Comment:
For confirmation - is validator part of this parameter definition? I'm coming from the m and efConstruction values validation, so they fit to what Lucene is supported. Or we're passing it to Lucene the relying on validation from them?

Reply:
Yes, we define validation logic here for parameters. We could remove it and rely on Lucene or we could match Lucene. What do you think?

Reply:
I see pros and cons of both but more inclined to have our own validation. If we declare support on our end provide documentation on what is supported and what's not it's became our burden to validate values. This came of course with the double-maintenance when we need to check on updates in Lucene and keep our code in sync, but I think such updates shouldn't happen frequently and can be done as part of upgrade to next core version.

Reply:
I think we add validation logic on our end as well. Looking at the code, the validation is the same as ours already: https://github.com/apache/lucene/blob/main/lucene/core/src/java/org/apache/lucene/util/hnsw/HnswGraphBuilder.java#L76